pkg/controller/command/credentialclient/command.go

```diff
@@ -52,6 +56,11 @@ func (c *Command) GetHandlers() []command.Handler {
 }
 
 func (c *Command) SaveCredential(_ io.Writer, req io.Reader) command.Error {
+	if c.sdsComm.SDSServerURL == "" {
+		logutil.LogError(logger, commandName, saveCredentialCommandMethod, failToStoreCredentialNoSDSURLSet)
+		return command.NewValidationError(InvalidRequestErrorCode, errFailToStoreCredentialNoSDSURLSet)
+	}
```

Comment:
This is just an extra layer of safety in case this method gets called somehow (perhaps by another dev in the future). Without this check, the error message is super confusing.

Reply:
We are not suppose to add check in every method

Reply:
Moved the check to a common spot in sdsComm.